recipes/qt/6.x.x/conanfile.py

```diff
@@ -242,6 +247,14 @@ def configure(self):
         if self.options.multiconfiguration:
             del self.settings.build_type
 
+        # If we are cross-compiling we need to enable these options primarily
+        #for the _native_ Qt build, because the builing Qt with the cross
+        # compiler requires the native Qt tools
+        if cross_building(self):
+            self.options["qt"].qttools = True
+            self.options["qt"].gui = True
+            self.options["qt"].widgets = True
+
```

Comment:
```suggestion
```

Reply:
Then I don't understand your commit message in https://github.com/conan-io/conan-center-index/pull/15678/commits/30a0fb315f03a44e7047bff977ba4030185bc64a

Reply:
Well, in the cross-compiling case, two instances of qt are built, because qt has itself as `tool_requires`.
The "tool_requires"-qt instance is built with the native tool-chain and requires the `qttools` submodule enabled.
When I set `-o qt:qttools=True` or `-o:b qt:qttools=True` on the command line of my `conan create` command it seems not to get through to the native qt build. Only the `qtbase` module is built in the native qt instance. But when I add `self.options["qt"].qttools = True` in `configure(self)` it works and the native `qttools` module gets compiled.

Reply:
please report this issue on conan : https://github.com/conan-io/conan/issues/new?assignees=&labels=&template=bug.yml&title=%5Bbug%5D+SHORT+DESCRIPTION

Regarding this PR, you have to know that setting option values in `configure(self)` is not enough to be sure that the option values are correct. The problem is that some other package in the dependency graph could change the option afterwards  (`self.options["qt"].qttools = False` for example). This is why you need to check the option values in `validate(self)`, because at this point you are sure that the option values won't change any more.